client/vault/vault.go

```diff
@@ -37,7 +36,6 @@ func NewClient(config Config) (*vaultapi.Client, error) {
 
 	newClientConfig := vaultapi.DefaultConfig()
 	newClientConfig.Address = address
-	newClientConfig.HttpClient = http.DefaultClient
```

Comment:
Calling `vaultapi.DefaultConfig()` already configures a HTTP client with transport support. So setting the HTTP client is not necessary and caused an error because `http.DefaultClient` doesn't configure transport support.

https://github.com/hashicorp/vault/blob/master/api/client.go#L89